File: 0892-surface-area-of-3d-shapes/0892-surface-area-of-3d-shapes.java
class Solution {
    public int surfaceArea(int[][] grid) {
        
        int rows = grid.length;
        int cols = grid[0].length;
        int area = 0;

        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {
                int h = grid[r][c];

                if (h > 0) {
                    area += 2;

                    area += (4 * h);

                    if (r > 0) {
                        area -= (2 * Math.min(h, grid[r - 1][c]));
                    }
                    if (c > 0) {
                        area -= (2 * Math.min(h, grid[r][c - 1]));
                    }
                }
            }

        }
        return area;
    }
}

File: 0892-surface-area-of-3d-shapes/0892-surface-area-of-3d-shapes.py
class Solution:
    def surfaceArea(self, grid: List[List[int]]) -> int:
        
        rows = len(grid)
        cols = len(grid[0])
        area = 0

        for r in range(rows):
            for c in range(cols):
                h = grid[r][c]
                if h > 0:
                    area += 2

                    area += 4 * h

                    if r > 0:
                        area -= 2 * min(h, grid[r - 1][c])
                    
                    if c > 0:
                        area -= 2 * min(h, grid[r][c - 1])
        return area
                